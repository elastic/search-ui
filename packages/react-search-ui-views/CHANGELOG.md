## 0.1.0 (January 3, 2019)

- Initial "Alpha" Release

## 0.2.0 (January 25, 2019)

- Updated type schema of `ValueFilterValue`.

## 0.4.0 (February 11, 2019)

- Updated styles

## 0.5.0 (March 6, 2019)

- #90
- #8
- #15
- #107
- #114

Breaking Changes:

- Facet component views have been renamed:
  - SingleRangeSelectFacet > SingleSelectFacet
  - MultiValueFacet > MultiCheckboxFacet
  - SingleValueLinksFacet > SingleLinksFacet
- The PropTypes validation changed for these views. If you were explicitly
  passing `options` or `values` parameters directly to these views, then you will
  likely have conflicts.
- Generally speaking the data model changed, especially with regards to `filters`, `facets`
  and `results`. If you were working directly with either of these types of data
  you will likely need to rework your solution. The data model can be seen
  here: https://github.com/elastic/search-ui/tree/8ddad47165d17c768a024a134059f215f9096365/packages/react-search-ui-views/src/types.

## 0.5.1 (March 7, 2019)

Fixed this error: "ReferenceError: regeneratorRuntime is not defined"
