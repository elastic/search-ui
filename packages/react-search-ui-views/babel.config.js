const presets = [
  [
    "@babel/env",
    {
      targets: {
        browsers: ["last 2 versions", "> 5%"]
      },
      modules: process.env.BABEL_MODULES
        ? process.env.BABEL_MODULES
        : "commonjs" // babel's default is commonjs
    }
  ],
  "@babel/preset-react"
];

module.exports = {
  sourceMaps: "inline",
  presets,
  plugins: ["@babel/plugin-proposal-class-properties"]
};
