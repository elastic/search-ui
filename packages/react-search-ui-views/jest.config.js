module.exports = {
  preset: "ts-jest/presets/js-with-ts-esm",
  testPathIgnorePatterns: ["./lib"],
  testEnvironment: "jsdom",
  setupFilesAfterEnv: ["<rootDir>/src/setupTests.js"],
  moduleNameMapper: {
    "\\.(css)$": "<rootDir>/__mocks__/styleMock.js"
  },
  snapshotSerializers: ["enzyme-to-json/serializer"]
};
