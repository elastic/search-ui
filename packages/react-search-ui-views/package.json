{
  "name": "@elastic/react-search-ui-views",
  "version": "0.4.0",
  "description": "A collection of React UI components for building search experiences",
  "license": "Apache-2.0",
  "main": "lib",
  "module": "es",
  "sideEffects": false,
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib",
    "es"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/elastic/search-ui.git"
  },
  "scripts": {
    "test": "jest",
    "clean": "rimraf es lib",
    "build-css": "./bin/build-css",
    "watch-css": "./bin/watch-css",
    "watch-js": "./bin/watch-js",
    "build-js": "./bin/build-js",
    "build": "run-s clean build-css build-js",
    "watch": "run-p watch-css watch-js",
    "prepare": "npm run build",
    "storybook": "start-storybook -p 6006",
    "build-storybook": "build-storybook"
  },
  "bugs": {
    "url": "https://github.com/elastic/search-ui/issues"
  },
  "peerDependencies": {
    "react": "^16.6.0"
  },
  "jest": {
    "setupTestFrameworkScriptFile": "<rootDir>/src/setupTests.js",
    "moduleNameMapper": {
      "\\.(css)$": "<rootDir>/__mocks__/styleMock.js"
    },
    "snapshotSerializers": [
      "enzyme-to-json/serializer"
    ]
  },
  "devDependencies": {
    "@babel/cli": "^7.1.2",
    "@babel/preset-react": "^7.0.0",
    "@storybook/addon-actions": "^4.1.0",
    "@storybook/addon-links": "^4.1.0",
    "@storybook/addons": "^4.1.0",
    "@storybook/react": "^4.1.0",
    "babel-loader": "^8.0.4",
    "chokidar-cli": "^1.2.1",
    "enzyme": "^3.7.0",
    "enzyme-adapter-react-16": "^1.6.0",
    "enzyme-to-json": "^3.3.4",
    "jest": "^23.6.0",
    "node-sass": "^4.11.0",
    "npm-run-all": "^4.1.5",
    "postcss-cli": "^6.1.1",
    "postcss-import": "^12.0.1",
    "react": "^16.6.0",
    "react-dom": "^16.6.0",
    "rimraf": "^2.6.2"
  },
  "dependencies": {
    "autoprefixer": "^9.4.7",
    "deep-equal": "^1.0.1",
    "rc-pagination": "^1.17.3",
    "react-select": "^2.1.1"
  }
}
