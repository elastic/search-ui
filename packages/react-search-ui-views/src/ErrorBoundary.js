import PropTypes from "prop-types";
import React from "react";

import { appendClassName } from "./view-helpers";

function ErrorBoundary({ children, className, error, ...rest }) {
  if (error) {
    return (
      <div className={appendClassName("sui-search-error", className)} {...rest}>
        {error}
      </div>
    );
  }

  return children;
}

ErrorBoundary.propTypes = {
  children: PropTypes.node.isRequired,
  error: PropTypes.string.isRequired,
  className: PropTypes.string
};

export default ErrorBoundary;
