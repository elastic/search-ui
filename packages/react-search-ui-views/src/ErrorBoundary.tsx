import React from "react";

import { appendClassName } from "./view-helpers";
import type { SearchContextState } from "@elastic/search-ui";
import { BaseContainerProps } from "./types";

export type ErrorBoundaryContainerContext = Pick<SearchContextState, "error">;

export type ErrorBoundaryViewProps = BaseContainerProps &
  ErrorBoundaryContainerContext;

function ErrorBoundary({
  children,
  className,
  error,
  ...rest
}: ErrorBoundaryViewProps & React.HTMLAttributes<HTMLDivElement>): JSX.Element {
  if (error) {
    return (
      <div className={appendClassName("sui-search-error", className)} {...rest}>
        {error}
      </div>
    );
  }

  return children as JSX.Element;
}

export default ErrorBoundary;
