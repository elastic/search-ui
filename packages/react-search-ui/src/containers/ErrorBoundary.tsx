import React, { Component } from "react";
import { withSearch } from "..";
import {
  ErrorBoundary,
  ErrorBoundaryContainerContext,
  ErrorBoundaryViewProps
} from "@elastic/react-search-ui-views";
import { BaseContainerProps } from "../types";

type ErrorBoundaryContainerProps = BaseContainerProps &
  ErrorBoundaryContainerContext & {
    view?: React.ComponentType<ErrorBoundaryViewProps>;
  };

export class ErrorBoundaryContainer extends Component<ErrorBoundaryContainerProps> {
  render() {
    const { children, className, error, view, ...rest } = this.props;

    const View = view || ErrorBoundary;

    const viewProps = {
      className,
      children,
      error,
      ...rest
    };

    return <View {...viewProps} />;
  }
}

export default withSearch<
  ErrorBoundaryContainerProps,
  ErrorBoundaryContainerContext
>(({ error }) => ({ error }))(ErrorBoundaryContainer);
