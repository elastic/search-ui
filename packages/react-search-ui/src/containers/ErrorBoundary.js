import PropTypes from "prop-types";
import { Component } from "react";
import { withSearch } from "..";
import { ErrorBoundary } from "@elastic/react-search-ui-views";

export class ErrorBoundaryContainer extends Component {
  static propTypes = {
    // Props
    children: PropTypes.node.isRequired,
    className: PropTypes.string,
    view: PropTypes.func,
    // State
    error: PropTypes.string.isRequired
  };

  render() {
    const { children, className, error, view, ...rest } = this.props;

    const View = view || ErrorBoundary;

    return View({
      className,
      children,
      error,
      ...rest
    });
  }
}

export default withSearch(({ error }) => ({ error }))(ErrorBoundaryContainer);
