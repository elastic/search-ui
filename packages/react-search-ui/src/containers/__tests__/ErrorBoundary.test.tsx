import React from "react";
import { shallow } from "enzyme";
import { ErrorBoundaryViewProps } from "@elastic/react-search-ui-views";
import { ErrorBoundaryContainer } from "../ErrorBoundary";

describe("Error Boundary", () => {
  const params = {
    children: <div>Child</div>,
    error: "I am an error"
  };

  let viewProps;

  const viewComponent: React.FC<ErrorBoundaryViewProps> = (
    props: ErrorBoundaryViewProps
  ) => {
    viewProps = props;
    return <div />;
  };

  beforeEach(() => {
    viewProps = null;
  });

  it("supports a render prop", () => {
    const render = ({ error }) => {
      return <div>{error}</div>;
    };
    const wrapper = shallow(
      <ErrorBoundaryContainer {...params} view={render} />
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });

  it("passes className through to the view", () => {
    const className = "test-class";
    shallow(
      <ErrorBoundaryContainer
        {...params}
        className={className}
        view={viewComponent}
      />
    ).dive();
    expect(viewProps.className).toEqual(className);
  });

  it("passes data-foo through to the view", () => {
    const data = "bar";
    shallow(
      <ErrorBoundaryContainer
        {...params}
        data-foo={data}
        view={viewComponent}
      />
    ).dive();
    expect(viewProps["data-foo"]).toEqual(data);
  });
});
