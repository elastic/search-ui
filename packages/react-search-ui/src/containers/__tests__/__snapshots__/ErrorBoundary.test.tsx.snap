// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Boundary supports a render prop 1`] = `
<div>
  I am an error
</div>
`;
